contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

```diff
@@ -405,14 +410,16 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
     if (consumers.length == MAX_CONSUMERS) {
       revert TooManyConsumers();
     }
-    mapping(uint256 => uint64) storage nonces = s_consumers[consumer];
-    if (nonces[subId] != 0) {
+    ConsumerConfig storage consumerConfig = s_consumers[consumer][subId];
+    if (consumerConfig.active) {
```

Comment:
Should we move this check before `if (consumers.length == MAX_CONSUMERS) {`, since attempting to add same consumer again should be an idempotent operation instead of reverting

Reply:
Yeah that makes sense technically. Can change it